src/shared/install.c

```diff
@@ -2868,15 +2908,73 @@ static int read_presets(UnitFileScope scope, const char *root_dir, Presets *pres
         return 0;
 }
 
-static int query_presets(const char *name, const Presets presets) {
+static int pattern_match_multiple_instances(
+                        const PresetRule rule,
+                        const char *unit_name,
+                        char ***ret) {
+
+        const char *parameter;
+        _cleanup_free_ char *templated_name = NULL;
+        int r;
+
+        /* If no ret is needed or the rule itself does not have instances
+         * initalized, we return not matching */
+        if (!ret || !rule.instances)
+                return 0;
+
+        r = unit_name_template(unit_name, &templated_name);
+        if (r < 0)
+                return r;
+        parameter = first_word(rule.pattern, templated_name);
```

Comment:
I'm confused now. Isn't `rule.pattern` a single word now?

Reply:
Sorry, that was a mistake from my side, will change it to `streq` instead. Thanks for pointing that out :).